File: cartracker/src/main/webapp/app/model/option/Category.js
Ext.define("CarTracker.model.option.Category",
{
  extend : "Ext.data.Model",
  fields : [ {
    name : "longName",
    type : "string"
  }, {
    name : "shortName",
    type : "string"
  }, {
    name : "createDate",
    type : "date",
    dateFormat : "c"
  }, {
    name : "active",
    type : "boolean",
    defaultValue : true
  }, {
    name : "id",
    type : "int",
    useNull : true,
    convert : null
  } ],
  proxy : {
    type : "direct",
    api : {
      read : "categoryOptionService.read",
      create : "categoryOptionService.create",
      update : "categoryOptionService.update",
      destroy : "categoryOptionService.destroy"
    },
    reader : {
      root : "records"
    }
  }
});

File: cartracker/src/main/java/ch/rasc/cartracker/entity/option/Category.java
package ch.rasc.cartracker.entity.option;

import javax.persistence.Entity;

import ch.rasc.extclassgenerator.Model;

@Entity
@Model(value = "CarTracker.model.option.Category",
		readMethod = "categoryOptionService.read",
		createMethod = "categoryOptionService.create",
		updateMethod = "categoryOptionService.update",
		destroyMethod = "categoryOptionService.destroy", paging = true)
public class Category extends BaseOption {
	// nothing special
}
